### PR TITLE
[ADD] HR Timesheet sheet attendance

### PR DESCRIPTION
**Pull request by bizzappdev**
Added new module  `hr_timesheet_sheet_attendance` to provide the missing functionality from V9 timesheet module

The module is providing timesheet Sheet integration with attendance.

### CODE REVIEW

**Comment by bizzappdev** [comment]
test cases are under development but other than that we can proceed for the code review and functional test.

File: hr_timesheet_sheet_attendance/__manifest__.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/__manifest__.py:7]: @bizzappdev Add `website` key, URL like [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L11)

File: hr_timesheet_sheet_attendance/models/hr_attendance.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/models/hr_attendance.py]: @bizzappdev should be `from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT, DEFAULT_SERVER_DATE_FORMAT`

File: hr_timesheet_sheet_attendance/models/hr_attendance.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/models/hr_attendance.py]: remove it

File: hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/models/hr_timesheet_sheet.py]: @bizzappdev `attendances_ids` string and `attendance_count` string is same, can you please change like `Attendance count`.

File: hr_timesheet_sheet_attendance/views/hr_attendance_view.xml
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/views/hr_attendance_view.xml]: remove it

File: hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/views/hr_timesheet_sheet_view.xml]: same here

**Review by nikul-serpentcs — COMMENTED** [review]
Improve Code

File: hr_timesheet_sheet_attendance/tests/__init__.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/tests/__init__.py]: add new blank line

File: hr_timesheet_sheet_attendance/tests/test_hr_attendance.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/tests/test_hr_attendance.py]: @bizzappdev IMO use `DEFAULT_SERVER_DATETIME_FORMAT`

File: hr_timesheet_sheet_attendance/tests/test_hr_attendance.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/tests/test_hr_attendance.py:30]: same here

File: hr_timesheet_sheet_attendance/__manifest__.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/__manifest__.py]: @bizzappdev remove it, Because `hr_timesheet_sheet` depends on `hr_timesheet`

File: hr_timesheet_sheet_attendance/__manifest__.py
Comment (nikul-serpentcs) [inline hr_timesheet_sheet_attendance/__manifest__.py:11]: same here ` hr_timesheet` depends on `hr_attendance`

**Review by nikul-serpentcs — CHANGES_REQUESTED** [review]
Minor Change

File: hr_timesheet_sheet_attendance/tests/test_hr_attendance.py
Comment (bizzappdev) [reply hr_timesheet_sheet_attendance/tests/test_hr_attendance.py:30]: yes. but we this method is doing same as it convert to datetime.object > change timezone > return string in odoo formate, that's why there is no need to convert it to DEFAULT_SERVER_DATETIME_FORMAT format

File: hr_timesheet_sheet_attendance/__manifest__.py
Comment (bizzappdevteam) [reply hr_timesheet_sheet_attendance/__manifest__.py:11]: hr_timesheet doesn't depend on hr_attendance
https://github.com/odoo/odoo/blob/11.0/addons/hr_timesheet/__manifest__.py#L23

Comment (nikul-serpentcs) [reply hr_timesheet_sheet_attendance/__manifest__.py:11]: @bizzappdevteam Ohhhh, I am checking `hr_timesheet_attendance`, keep it.

I check [here](https://github.com/odoo/odoo/blob/11.0/addons/hr_timesheet_attendance/__manifest__.py#L12)

**Comment by bizzappdev** [comment]
test cases are also been added. sorry, forget to mention it before.

**Comment by bizzappdev** [comment]
@nikul-serpentcs changes are done. Can you please review again.
we need another reviewer also if you can ping someone then it would be nice

**Review by nikul-serpentcs — APPROVED** [review]
Code Review LGTM :+1:

**Comment by nikul-serpentcs** [comment]
> @nikul-serpentcs changes are done. Can you please review again.
> we need another reviewer also if you can ping someone then it would be nice

@pedrobaeza @alexey-pelykh @aheficent Can you please take a look?

**Review by AaronHForgeFlow — COMMENTED** [review]
Is it possible to not allow to check in / out from timesheet form view? It is shrinking the list of attendances and it does not help much IMO.

**Comment by bizzappdev** [comment]
IMO it is a nice feature that user can manipulate and change/add from timesheet view itself.
since V9 it was also like this. 
as far as shrinking the list of attendance is a concern we have only check-in and checkout DateTime field only in that view. still, it is a debatable topic.

**Comment by AaronHForgeFlow** [comment]
It is more about the alignment. The actions are not aligned to the current status of the attendance. I know it is a minor thing and I am being picky but it'd look better with that, maybe the action could be below the current status.

image

**Comment by bizzappdev** [comment]
indeed it should be aligned. and nice to have an idea about below the Status. we will do the changes

**Review by AaronHForgeFlow — APPROVED** [review]
Great! Thanks. It works on runbot.

**Pull request merged**
